python_modules/libraries/dagster-components/dagster_components/core/component_key.py

```diff
@@ -41,11 +53,18 @@ def from_typename(typename: str) -> "GlobalComponentKey":
 class LocalComponentKey(ComponentKey):
     dirpath: Path
 
+    def __post_init__(self) -> None:
```

Comment:
can we switch to record so we don't have to do this crap

Reply:
`@record` lives in `dagster` so we can't depend on it in dagster-dg :/